src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/LanguageServices/CSharpLanguageServiceHost.cs

```diff
@@ -25,14 +26,11 @@ protected override Guid IntelliSenseProviderGuid
             get { return CSharpIntellisenseProvider; }
         }
 
-        /// <summary>
-        /// Invoked when the UnconfiguredProject is first loaded to initialize language services.
-        /// </summary>
-        [ProjectAutoLoad(ProjectLoadCheckpoint.InitialActiveConfigurationKnown)]
+        [ProjectAutoLoad(ProjectLoadCheckpoint.ProjectFactoryCompleted)]
         [AppliesTo(ProjectCapability.CSharpLanguageService)]
-        private void Initialize()
+        private Task OnProjectFactoryCompleted()
         {
-            var nowait = InitializeAsync();
+            return InitializeAsync();
```

Comment:
If we await it, we have control over scheduling the continuation. In this case, since we don't know the caller's sync context, we'd likely want to do await InitializeAsync().ConfigureAwait(false). If we simply pass the task through we don't control that. It might still be fine as the caller of OnProjectFactoryCompleted might be doing the right thing but maybe not.

Also if the caller doesn't await the task, the exception semantics maybe different - not sure.